src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -0,0 +1,471 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Model;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+[assembly: LoadableClass(typeof(NormalizerTransformer), null, typeof(SignatureLoadModel),
+    "", NormalizerTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(NormalizerTransformer), null, typeof(SignatureLoadRowMapper),
+    "", NormalizerTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public sealed class Normalizer : IEstimator<NormalizerTransformer>
+    {
+        private static class Defaults
+        {
+            public const bool FixZero = true;
+            public const bool MeanVarCdf = false;
+            public const bool LogMeanVarCdf = true;
+            public const int NumBins = 1024;
+            public const int MinBinSize = 10;
+            public const long MaxTrainingExamples = 1000000000;
+        }
+
+        public enum NormalizerMode
+        {
+            MinMax = 0,
+            MeanVariance = 1,
+            LogMeanVariance = 2,
+            Binning = 3
+        }
+
+        public abstract class ColumnBase
+        {
+            public readonly string Input;
+            public readonly string Output;
+            public readonly long MaxTrainingExamples;
+
+            protected ColumnBase(string input, string output, long maxTrainingExamples)
```

Comment:
>protected [](start = 12, length = 9)

Is there any reason not to make this and all these new public abstract classes you're creating subclassable by literally anyone? I don't see so, at least in the code written here. So how about a nice `private protected` to ease your conscience? :)